### PR TITLE
doc: document existing community features

### PR DESCRIPTION
**Pull request by bobtwinkles**
[rendered](https://github.com/bobtwinkles/DelugeFirmware/blob/doc/features.md/CommunityFeatures.md)

### CODE REVIEW

File: CommunityFeatures.md [hunk 1]
```diff
@@ -1,40 +1,76 @@
...
+# File Compatibility Warning
+In general, we try to maintain file compatibility with the official firmware. However, **files (including songs, presets, etc.) that use community features may not ever load correctly on the official firmware again**. Make sure to back up your SD card!
+
+# General improvements
+
+Here is a list of general improvements that have been made ordered from newest to oldest:
+
+* PR [#17] - Increase the resolution of "patch cables" between mod sources and destinations.
+* PR [#29] - Distinguish between MPE and non-MPE kit rows.
+* PR [#47] - Extra MIDI ports on the USB interface for MPE. This improves the usability of MPE-capable devices through the USB interface by providing more virtual cables.
```
Comment (bobtwinkles) [inline CommunityFeatures.md:14]: @m-m-adams I think this is correct, but please double check the description for #47 and #49.

File: CommunityFeatures.md [hunk 2]
```diff
@@ -1,40 +1,76 @@
...
+
+Here is a list of features that have been added to the firmware as a list ordered from newest to oldest:
+
+## Synthesizer features
+
+### New LFO shapes
+LFO types added to the "LFO SHAPE" shortcut.
+
+ - ([#32]) Random Walk. Starts at zero and walks up or down by small increments when triggered.
+ - ([#32]) Sample&Hold. Picks a new random value every time it is triggered.
```
Comment (bobtwinkles) [inline CommunityFeatures.md:27]: @phfalk please double check this documentation is correct (and for LFOF sync modes below).

File: CommunityFeatures.md [hunk 3]
```diff
@@ -1,40 +1,76 @@
...
+
+### New LFO Synchronization Modes
+Synchronization modes accessible through the "LFO SYNC" shortcut.
+
+ - ([#32]) Triplets. Synchronizes the LFO to triplet (3/2) divisions.
+ - ([#32]) Dotted. Synchronizes the LFO to "dotted" (2/3) divisions.
+
+### Filters
+ - ([#103]) adds a new filter in the low-pass slot, a state-variable filter.
```
Comment (bobtwinkles) [inline CommunityFeatures.md:36]: @m-m-adams please confirm.

File: CommunityFeatures.md [hunk 4]
```diff
@@ -1,40 +1,76 @@
...
+
+## New behaviors
+
+### Instrument Keyboard View
+ - ([#46]) Note offset between rows is now configurable by holding shift and using the horizontal encoder. This allows e.g. an isomorphic keyboard layout by setting the row offset to 12. The setting is saved per clip in the song file.
```
Comment (bobtwinkles) [inline CommunityFeatures.md:41]: @bfredl let me know if there's anything else to note here.

File: CommunityFeatures.md [hunk 5]
```diff
@@ -1,40 +1,76 @@
...
+
+### Kit Clip View
+ - ([#122]) Pressing "AUDITION + RANDOM" on a drum kit row will load a random sample from the same folder as the currently enabled sample and load it as the sound for that row. Currently limited to 25 files for performance reasons. This feature can be toggled in the [runtime features menu](#runtime-features).
```
Comment (bobtwinkles) [inline CommunityFeatures.md:44]: @alter-alter to confirm I've got the behavior right

File: CommunityFeatures.md [hunk 6]
```diff
@@ -1,40 +1,76 @@
...
+
+### Kit Keyboard View
+ - ([#112]) All-new use for the "keyboard" button in kit clips, uses the main pad grid for MPC-style 16 level playing.
```
Comment (bobtwinkles) [inline CommunityFeatures.md:47]: @alter-alter / @m-m-adams. Should we document scroll behavior here too?

File: CommunityFeatures.md (hunk 4 above)
Comment (bfredl) [reply CommunityFeatures.md:41]: I think the keyboard is isomorphic in any case, perhaps "e.g. an octave keyboard layout by setting the row offset to 12"

File: CommunityFeatures.md (hunk 2 above)
Comment (phfalk) [reply CommunityFeatures.md:27]: I am a little late on this and the PR is already merged, but I have a small correction here: The initial value for the random walk isn't zero but a random value within +/- range/2. What is the process to change this? @jamiefaye, do you want me to make another PR?

**Pull request merged**